packages/json-rpc-middleware-stream/tsconfig.json

```diff
@@ -0,0 +1,13 @@
+{
+  "extends": "../../tsconfig.packages.json",
+  "compilerOptions": {
+    "baseUrl": "./",
+    "rootDir": "../..",
+    "exactOptionalPropertyTypes": true,
+    "forceConsistentCasingInFileNames": true,
+    "noErrorTruncation": true,
+    "noUncheckedIndexedAccess": true,
+    "target": "es2020"
+  },
+  "include": ["../../types", "../../tests", "./src", "./tests"]
```

Comment:
Consulting the [original repo](https://github.com/MetaMask/json-rpc-middleware-stream/blob/main/tsconfig.json), it doesn't look like `exactOptionalPropertyTypes`, `noErrorTruncation, or `noUncheckedIndexedAccess` are present, and the `target` seems to be set to `es2017` (instead of `es6`, which is the default for this repo). `skipLibCheck` _is_ set, but we could drop that if it's not causing any problems.

Additionally, consulting other packages in this repo, e.g. [`accounts-controller`](https://github.com/MetaMask/core/blob/5ff1505aa4833f7b6c9055cc734c18be2401b661/packages/accounts-controller/tsconfig.json), `rootDir` doesn't need to be set, and `include` doesn't need to include `tests`.

What do you think about:

``` suggestion
  "compilerOptions": {
    "baseUrl": "./",
    "forceConsistentCasingInFileNames": true,
    "target": "es2017"
  },
  "include": ["../../types", "./src"]

Reply:
I agree, we should only be preserving the compiler options of the original, not introducing new ones.

Reply:
Ah. Good eye! That is because they reference type augmentations — type overrides for packages — that are defined _outside_ of the package. If we didn't define `"rootDir": "../.."` in those cases then TypeScript would give us an error. In this case, it doesn't look like we are accessing those types, so we don't need to specify a `rootDir`. Does that make sense? (Perhaps we ought to be more consistent about this.)